tests/data_result/module_130_140/models/res_partner.py

```diff
@@ -0,0 +1,58 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+    
+    def test_improve_translation(self):
+        return _("It's %s", 2024)
+    
+    def test_improve_transalation_with_line_break(self):
+        raise ValidationError(
+            _("The '%s' is empty or 0. It should have a non-null value.", "Price")
+        )
+    
+    def test_improve_translation_with_parenthesis(self):
+        return _("It's %s", 2024)
+    
+    def test_improve_translation_with_single_quote(self):
+        return _('It is %s', 2024)
+    
+    def test_improve_transalation_with_parenthesis_and_line_break(self):
+        raise ValidationError(
+            _("%s are only valid until %s", "User", 2024)
```

Comment:
my problem with this one is that it's still problematic: it contains 2 unnamed placeholders

> as a best practiace, we should not use multiple unnamed placeholders in a translated string. In some languages the order of the placeholders may have to be modified, which is impossible if they are unnamed. We can use named placedholders to avoid this

can you think about this case? maybe a warning?